pkg/ring/kv/consul/client.go

```diff
@@ -62,6 +65,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, prefix string) {
 	f.StringVar(&cfg.ACLToken, prefix+"consul.acltoken", "", "ACL Token used to interact with Consul.")
 	f.DurationVar(&cfg.HTTPClientTimeout, prefix+"consul.client-timeout", 2*longPollDuration, "HTTP timeout when talking to Consul")
 	f.BoolVar(&cfg.ConsistentReads, prefix+"consul.consistent-reads", true, "Enable consistent reads to Consul.")
+	f.Float64Var(&cfg.WatchKeyRate, prefix+"consul.watch-rate", 0, "Rate limit when watching key or prefix in Consul, in requests per second. Zero disables the rate limit.")
+	f.IntVar(&cfg.WatchKeyBurst, prefix+"consul.watch-burst", 1, "Burst size used in rate limit. Values less than 1 are treated as 1.")
```

Comment:
I'd use `watch-burst-size` maybe?

Reply:
Technically it's the max number of tokens you can consume in a single request. Given here you always consume 1, I'm wondering if it it's really useful to allow people to configure it, instead of simply hardcoding it to `1`.